pandas/core/generic.py

```diff
@@ -1084,6 +1084,10 @@ def to_hdf(self, path_or_buf, key, **kwargs):
                        / selecting subsets of the data
         append   : boolean, default False
             For Table formats, append the input data to the existing
+        data_columns : list of columns, or True, default None
+            This will create additional indexed columns for on-disk queries,
+            by default only 'index' and 'columns' are indexed. True will index
+            all columns.
```

Comment:
this is a generic doc-string that applies when you are looknig at Series,DataFrame, or Panel. so it should either be somewhat generic or specific to that type (which is harder and not worth it). You are saying 'index' and 'columns', which don't apply to Series/Panel.